sklearn/cross_validation.py

```diff
@@ -80,16 +92,36 @@ def indices(self):
         return self._indices
 
     def __iter__(self):
+        return self.split(None)
+
+    def split(self, y):
```

Comment:
Data is X. y isn't data, is it? But this would also be applied to y, right? Or either? Sorry, I haven't read enough of the rest of the PR.